R/pkgs/hospitalization/R/hospdeath.R

```diff
@@ -78,68 +251,321 @@ build_hospdeath_par <- function(p_hosp, p_death, p_ICU, p_vent, data_filename, s
+    NULL
+  }
+  doParallel::stopImplicitCluster()
+}
+
+
+##'
+##' Build a set of sampled hospitalizations, deaths, and recoveries
+##' from the incident infection data from the simulation model.
+##' Use GEOID-specific, age-adjust probabilities of hospitalization,
+##' death, and ICU admission
+##'
+##' @param prob_dat df of p_hosp, p_death, p_ICU, p_vent, GEOID
+##' @param scl_fac scaling factor for the probability symptomatic infection
+##' @param data_dir Path to the directory that contains the CSV output of the simulation model
+##' @param dscenario_name The name of the scenario we are analyzing here (e.g., "highdeath", "meddeath", etc.)
+##' @param time_hosp_pars parameters for time from onset to hospitalization distribution
+##' @param time_ICU_pars parameters for time from hospitalization to ICU
+##' @param time_vent_pars parameters for time from ICU to time on ventilator
+##' @param time_onset_death_pars parameters for time from onset to death distribution
+##' @param time_disch_pars parameters for time from hospitalization to discharge parameters
+##' @param time_ICUdur_pars parameetrs for time of ICU duration
+##' @param cores The number of CPU cores to run this model on in parallel
+##' @param root_out_dir Path to the directory to write the outputs of this analysis
+##'
+##' @export
+build_hospdeath_geoid_par <- function(
+  prob_dat,
+  scl_fac,
+  data_dir,
+  dscenario_name,
+  time_symp_pars = c(log(5.2), log(1.74)),
+  time_hosp_pars = c(1.23, 0.79),
+  time_ICU_pars = c(log(10.5), log((10.5-7)/1.35)),
+  time_vent_pars = c(log(10.5), log((10.5-8)/1.35)),
+  time_onset_death_pars = c(log(11.25), log(1.15)),
+  time_disch_pars = c(log(11.5), log(1.22)),
+  time_ICUdur_pars = c(log(17.46), log(4.044)),
+  time_ventdur_pars = log(17),
+  cores=8,
+  root_out_dir='hospitalization',
+  use_parquet = FALSE) {
+
+  ## add in scaling factor to p_symp_inf
+  prob_dat$p_symp_inf_scaled <- prob_dat$p_symp_inf * scl_fac
+  prob_dat$p_symp_inf_scaled[prob_dat$p_symp_inf_scaled>1] <- 1
+
+  n_sim <- length(list.files(data_dir))
+  print(paste("Creating cluster with",cores,"cores"))
+  doParallel::registerDoParallel(cores)
+
+  if(n_sim == 0){
+    stop("No simulations selected to run")
+  }
+
+  print(paste("Running over",n_sim,"simulations"))
+
+  pkgs <- c("dplyr", "readr", "data.table", "tidyr", "hospitalization")
+  foreach::foreach(s=seq_len(n_sim), .packages=pkgs) %dopar% {
+    dat_I <- hosp_load_scenario_sim(data_dir,s,
+                                    keep_compartments = "diffI",
+                                    geoid_len=5,
+                                    use_parquet = use_parquet) %>%
+      mutate(hosp_curr = 0,
+             icu_curr = 0,
+             vent_curr = 0,
+             uid = paste0(geoid, "-",sim_num)) %>%
+      rename(incidI = N)
+    dat_ <- left_join(dat_I, prob_dat, by="geoid")
+
+    # Add time things
+    dat_symp <- hosp_create_delay_frame('incidI', dat_$p_symp_inf_scaled, dat_,
```

Comment:
@salauer should confirm - but this is actually wrong. There should be no delay from incident I to incident Symptomatic (i.e. E class already encompasses the incubation period). Again, would just remove entire function

Reply:
yeah, I agree. That seems out of place. Remove line 85 (not line 82)

Reply:
this is wrong but also is deleted now in dataseed_merge